packages/lesswrong/lib/collections/posts/permissions.ts

```diff
@@ -44,6 +46,8 @@ Posts.checkAccess = async (currentUser: DbUser|null, post: DbPost, context: Reso
     return true
   } else if (userOwns(currentUser, post) || userIsSharedOn(currentUser, post) || await userIsPostGroupOrganizer(currentUser, post)) {
     return true;
+  } else if (!currentUser && constantTimeCompare(getSharingKeyFromContext(context), post.linkSharingKey)) {
```

Comment:
Huh, I guess `getSharingKeyFromContext` does work. My belief that this was going to be hard was partially predicated on the belief that `checkAccess` didn't have a reasonable way to see whether a link-sharing key was provided.

This is still pretty dodgy, since `context` is unavailable client side, and additionally, is unavailable if we're in the context of a GraphQL made separately from the page's main SSR. I think what that means in practice is that if you navigate to a link-shared post rather than going there directly in a new tab, it doesn't work. OTOH if you're logged out, you can't be following a link inside a DM, so... maybe it works out okay?

I have a pretty strong intuition that this is going to fail *somewhere* if we QA it thoroughly. But I could be wrong, and if we don't actually find any problems maybe it's okay. If there are problems, the fix I'd look at would be to extend `linkSharingKeyUsedBy` to handle `clientId`s (which logged-out users have, and which are in a cookie on the `req` which the server can access even if this isn't an SSR).

Reply:
Yeah, I agree that this is kind of dodgy. It does seem worth trying out what exactly happens and what is provided when we make a graphQL query outside of the main SSR page, and seeing whether there is maybe still a way to make it work.

I was thinking about the `linkSharingKeyUsedBy` modification to handle clientIds, but was kind of worried, since IIRC we just provide the clientIds directly as cookies, and they are much shorter, so brute-forcing one might not be out of the question (at least compared to tokens). Now thinking a bit more about it, it probably is still fine, but it does feel a bit like a headache.